tests/cpp/test_serial_gridreduce.cpp

```diff
@@ -37,110 +37,91 @@ using SerialGridReductionTest = NVFuserTest;
-
-          // Unreduced dimensions should be concrete. Reduced dimension could be
-          // symbolic, but is concrete here so that we can read tv0 to registers
-          TensorView* tv0 = TensorViewBuilder()
-                                .shape({H, W})
-                                .dtype(DataType::Float)
-                                .contiguity(true)
-                                .build();
-          fusion->addInput(tv0);
-
-          auto tv1 = sum(tv0, {0});
-          fusion->addOutput(tv1);
-
-          // Start with
-          //   [ rS{H}, iS{W} ]
-          // We are grid reducing the H dimension and we want to coalesce
-          // accesses in the W dimension. So we first reorder to
-          //   [ iS{W}, rS{H} ]
-          // then schedule as
-          //   [ iBIDx{blocks_x}, iBIDy{blocks_y}, iS{A}, iS{B},
-          //   iTIDy{num_warps}, iTIDx{32}, rBIDz{blocks_z} ]
-          auto tv2 = tv0->cacheAfter();
-          auto tv3 = tv1->cacheBefore();
-
-          tv3->reorder(
-              {{1, 0}, {0, 1}}); // blocks_x*blocks_y*A*B*num_warps*32, H
-          tv3->split(0, 32); // blocks_x*blocks_y*A*B*num_warps, 32, H
-          tv3->split(0, num_warps); // blocks_x*blocks_y*A*B, num_warps, 32, H
-          tv3->split(0, B); // blocks_x*blocks_y*A, B, num_warps, 32, H
-          tv3->split(0, A); // blocks_x*blocks_y, A, B, num_warps, 32, H
-          tv3->split(0, blocks_y); // blocks_x, blocks_y, A, B, num_warps, 32, H
-          tv3->axis(0)->parallelize(ParallelType::BIDx);
-          tv3->axis(1)->parallelize(ParallelType::BIDy);
-          tv3->axis(4)->parallelize(ParallelType::TIDy);
-          tv3->axis(5)->parallelize(ParallelType::TIDx);
-          tv3->axis(6)->parallelize(ParallelType::BIDz);
-          // Reorder to put parallel dims first for better inlining
-          tv3->reorder({
-              {4, 2},
-              {5, 3},
-              {2, 4},
-              {3, 5},
-          });
-
-          TransformPropagator propagator(tv3);
-          MaxRootDomainInfoSpanningTree(tv3).traverse(&propagator);
-          scheduler_utils::parallelizeAllLike(tv3);
-
-          // Here we just transpose A and B in tv2, so that it will be partially
-          // inlined with tv3, resulting in a separate loop to load tv0 into
-          // registers (tv2).
-          tv2->reorder({
-              {-2, -3},
-              {-3, -2},
-          });
-
-          inlineMost();
-
-          FusionExecutor fe;
-          if (serial) {
-            tv3->definition()->as<ReductionOp>()->requestSerialGridReduction();
-          }
-          fe.compileFusion(fusion);
-
-          auto input = at::randn(
-              {H, W},
-              at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0));
-          auto outputs = fe.runFusion({input});
-
-          if (serial) {
-            testValidate(fusion, outputs, {input}, __LINE__, __FILE__);
-          }
+    for (int64_t num_warps : {4, 8}) {
```

Comment:
This test doesn't seem to add much value unless it validates no explicit zero clear is done after each call. Otherwise, it seems like just a normal grid reduction test.

Reply:
You're right that this test has less value now that serial grid reduction is used in matmul split-K. We could remove it. It's difficult to test whether a memset kernel was launched. I could test that there is some zeroed memory allocated after this call, but that will only really be informative when this test is run in isolation since other tests that are run earlier could cause that allocation also.

Reply:
Maybe capturing the `global_zeroed_memory` output and doing some string match? I remember gtest has some utility function to do that.

Reply:
Added